src/Microsoft.VisualStudio.ProjectSystem.VisualBasic.VS/ProjectSystem/VS/Automation/VSImports.cs

```diff
@@ -0,0 +1,163 @@
+                    var importsList = project.GetItems(importItemTypeName)
+                                             .Select(p => p.EvaluatedInclude)
+                                             .ToList();
+                    return importsList.ElementAt(lIndex - 1);
+                }
+            });
+        }
+
+        public void Add(string bstrImport)
+        {
+            _threadingService.ExecuteSynchronously(async () =>
+            {
+                using (var access = await _lockService.WriteLockAsync())
+                {
+                    var project = await access.GetProjectAsync(ConfiguredProject).ConfigureAwait(true);
+                    await access.CheckoutAsync(project.Xml.ContainingProject.FullPath).ConfigureAwait(true);
+                    project.AddItem(importItemTypeName, bstrImport);
+                }
+            });
+        }
+
+        public void Remove(object index)
+        {
+            _threadingService.ExecuteSynchronously(async () =>
+            {
+                using (var access = await _lockService.WriteLockAsync())
+                {
+                    var project = await access.GetProjectAsync(ConfiguredProject).ConfigureAwait(true);
+                    await access.CheckoutAsync(project.Xml.ContainingProject.FullPath).ConfigureAwait(true);
+                    var importProjectItem = project.GetItems(importItemTypeName)
+                                                   .ElementAt(((int)index - 1));
+                    project.RemoveItem(importProjectItem);
+                }
+            });
+        }
+
+        public IEnumerator GetEnumerator()
+        {
+            return _threadingService.ExecuteSynchronously(async () =>
+            {
+                using (var access = await _lockService.ReadLockAsync())
+                {
+                    var project = await access.GetProjectAsync(ConfiguredProject).ConfigureAwait(true);
+                    return project.GetItems(importItemTypeName)
+                                  .Select(p => p.EvaluatedInclude)
+                                  .GetEnumerator();
+                }
+            });
+        }
+
+        public DTE DTE => _vsProject.DTE;
+
+        public Project ContainingProject => _vsProject.Project;
+
+        public int Count
+        {
+            get
+            {
+                return _threadingService.ExecuteSynchronously(async () =>
+                {
+                    using (var access = await _lockService.ReadLockAsync())
+                    {
+                        var project = await access.GetProjectAsync(ConfiguredProject).ConfigureAwait(true);
+                        return project.GetItems(importItemTypeName).Count;
+                    }
+                });
+            }
+        }
+
+        public object Parent => throw new NotImplementedException();
+
+        internal virtual void OnImportAdded(string importNamespace)
+        {
```

Comment:
My bad, this should have been raised when the import was added and the reference page will be called back. For the scenario of adding or removing the import from the list, this will be a no-op because we have a flag that is set in the reference page which says do not do anything when the page is adding or removing imports. 

This event addresses the scenario where the property page is open(and active) but the imports are updated programmatically. 

I will address this.